### PR TITLE
PE-130: API URL change

### PR DESCRIPTION
**Pull request by tbansal-opennms**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-130

### CODE REVIEW

File: ui/src/components/Common/Demo/apiInterceptor.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import axios from 'axios'
-import { BASE_REST_URL, BASE_REST_AUTH } from '../../../../devConfig'
+//import { BASE_REST_URL, BASE_REST_AUTH } from '../../../../devConfig'
 
 export const axiosAuth = axios.create()
```
Comment (mikewrosey) [inline ui/src/components/Common/Demo/apiInterceptor.ts:4]: If axiosAuth is no longer used anywhere, should be able to delete this file and devConfig.

File: ui/src/services/configurationService.ts [hunk 2]
```diff
@@ -1,44 +1,18 @@
...
-    }
-  } catch {
-    console.error('issue with getAdvancedDropdown api')
-  }
-})
-
-const getProvisionDService = axiosAuth.get(getProvisionD).then((response) => {
+const getDropdownTypes = typesUrl;
+const getSchedulePeriod = periodUrl;
+const getAdvancedDropdown = advDropdownUrl;
```
Comment (mikewrosey) [inline ui/src/services/configurationService.ts:12]: Have you tried importing with the variable name you want to use? something like:
import getDropdownTypes from '../../src/components/Common/Demo/MockupData/types.json'
That way they don't have to be reassigned and we can remove three lines here.

File: ui/src/components/Common/Demo/apiInterceptor.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import axios from 'axios'
-import { BASE_REST_URL, BASE_REST_AUTH } from '../../../../devConfig'
+//import { BASE_REST_URL, BASE_REST_AUTH } from '../../../../devConfig'
```
Comment (mikewrosey) [inline ui/src/components/Common/Demo/apiInterceptor.ts:2]: If axiosAuth is no longer used anywhere, should be able to delete this file and devConfig.

**Comment by mikewrosey** [comment]
Merging CM-Phase1 into this and resolving conflicts should fix the ui build. Let me know if you need help with that.

File: ui/src/services/configurationService.ts (hunk 2 above)
Comment (tbansal-opennms) [reply ui/src/services/configurationService.ts:12]: Done

File: ui/src/components/Common/Demo/apiInterceptor.ts (hunk 3 above)
Comment (tbansal-opennms) [reply ui/src/components/Common/Demo/apiInterceptor.ts:2]: For now we can't remove these files as these are helpfull to develop without VM. Will remove these files after all feather components development.

File: ui/src/components/Configuration/ReqDefinitionForm.vue [hunk 4]
```diff
@@ -280,53 +280,85 @@ const generateURL = () => {
...
+        }
+    } catch {
+        notify({
+            title: "Notification",
+            text: 'ProvisionDService PUT API Error',
+            type: 'error',
+        });
+    }
+    }
+};
```
Comment (mikewrosey) [inline ui/src/components/Configuration/ReqDefinitionForm.vue:360]: Formatting should follow the rules in the prettierrc file, but can do a bulk fix later

**Review by mikewrosey — APPROVED** [review]

**Pull request merged**